src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,473 @@
+  width="100%" >
+</iframe>
+
+Here's why the preceding example fails to print the result of `getUserOrder()`:
+
+* `getUserOrder()` is an asynchronous function that, after a delay, 
+eventually provides a string that describes the user's order: a "Large Latte".
+* To get the user's order, `createOrderMessage()` should call `getUserOrder()`
+and wait for it to finish after a delay. Because `createOrderMessage()` does *not* wait 
+for `getUserOrder()` to finish, `createOrderMessage()` fails to get the string value
+that `getUserOrder()` eventually provides. 
+* Instead, `createOrderMessage()` gets a representation of pending work to be
+done: an uncompleted future. You'll learn more about futures in the next section.
+* Because `createOrderMessage()` fails to get the value describing the user's 
+order, the example fails to print "Large Latte" to the console, and instead 
+prints "Your order is: Instance of '_Future<String>'".
+
+In the next sections you'll learn the about futures, `async`, and `await`
+so that you'll be able to write the code necessary to make `getUserOrder()` 
+print the desired value ("Large Latte") to the console.
+
+ {{ site.alert.secondary }} 
+  **Key terms:** 
+
+  * **synchronous operation**: A synchronous operation blocks other operations from executing until it completes. 
+
+  * **synchronous function**: A synchronous function only performs synchronous operations.
+
+  * **asynchronous operation**: Once initiated, an asynchronous operation allows 
+    other operations to execute before it completes.
+
+  * **asynchronous function**: An asynchronous function performs at least one 
+  asynchronous operation and can also perform _synchronous_ operations. 
+
+ {{ site.alert.end }}
+
+
+## What is a future?
+A future (lower case "f") is an instance of the [Future][future class]
+(capitalized "F") class. A future represents the result of an asynchronous 
+operation, and can have two states: uncompleted or completed.
+
+{{ site.alert.note }}
+  _Uncompleted_ is a Dart term referring to the state of a future
+  before it has produced a value.
+{{ site.alert.end }}
+
+### Uncompleted
+
+When you call an asynchronous function, it returns an uncompleted future. 
+That future is waiting for the function's asynchronous operation to finish or to throw an error. 
+
+### Completed
+If the asynchronous operation succeeds, the future completes with a 
+value. Otherwise it completes with an error.
+
+#### Completing with a value
+
+A future of type `Future<T>` completes with a value of type `T`.
+For example, `Future<String>` is the type signature for a
+future that produces a string value. If a future doesn't produce a usable value
+– for example, if you only need to log that a task is complete – then the future's type is `Future<void>`.
+
+#### Completing with an error
+
+If the asynchronous operation performed by the function fails for any reason, the 
+future completes with an error.
+
+### Example: Introducing futures
```

Comment:
I'm happy to riff on other possible names: 
```
Example: 
1. Fun with Futures
2. Futures Fundamentals
3. The basics of Futures
4. Facts about Futures
5. Back to the Futures - this would require re-working the example to use a lot of "Maaaartty" or "MCFLY!!" strings. Why didn't we notice this sooner? I regret everything.